core/src/main/java/org/apache/calcite/sql/type/ArrayElementOperandTypeChecker.java

```diff
@@ -34,61 +33,64 @@
 public class ArrayElementOperandTypeChecker implements SqlOperandTypeChecker {
   //~ Instance fields --------------------------------------------------------
 
-  private final boolean allowNullCheck;
-  private final boolean allowCast;
+  private final boolean arrayMayBeNull;
+  private final boolean elementMayBeNull;
 
   //~ Constructors -----------------------------------------------------------
 
-  public ArrayElementOperandTypeChecker() {
-    this.allowNullCheck = false;
-    this.allowCast = false;
-  }
-
-  public ArrayElementOperandTypeChecker(boolean allowNullCheck, boolean allowCast) {
-    this.allowNullCheck = allowNullCheck;
-    this.allowCast = allowCast;
+  public ArrayElementOperandTypeChecker(boolean arrayMayBeNull, boolean elementMayBeNull) {
+    this.arrayMayBeNull = arrayMayBeNull;
+    this.elementMayBeNull = elementMayBeNull;
   }
 
   //~ Methods ----------------------------------------------------------------
 
   @Override public boolean checkOperandTypes(
       SqlCallBinding callBinding,
       boolean throwOnFailure) {
```

Comment:
If array is not first argument? such as array_xxx(1, array, 2). It seems that this typechecker only works in this situation.